libshaderc/src/shaderc_cpp_test.cc

```diff
@@ -72,121 +127,75 @@ TEST(CppInterface, MultipleThreadsInitializing) {
   EXPECT_TRUE(compiler3->IsValid());
 }
 
-// Compiles a shader and returns true on success, false on failure.
-bool CompilationSuccess(const shaderc::Compiler& compiler,
-                        const std::string& shader, shaderc_shader_kind kind) {
-  return compiler.CompileGlslToSpv(shader.c_str(), shader.length(), kind)
-      .GetSuccess();
-}
-
-// Compiles a shader and returns true if the result is valid SPIR-V.
-bool CompilesToValidSpv(const shaderc::Compiler& compiler,
-                        const std::string& shader, shaderc_shader_kind kind) {
-  shaderc::SpvModule result = compiler.CompileGlslToSpv(shader, kind);
-  if (!result.GetSuccess()) return false;
-  size_t length = result.GetLength();
-  if (length < 20) return false;
-  const uint32_t* bytes =
-      static_cast<const uint32_t*>(static_cast<const void*>(result.GetData()));
-  return bytes[0] == spv::MagicNumber;
-}
-
-// Compiles a shader with the given options and returns true if the result is
-// valid SPIR-V.
-bool CompilesToValidSpv(const shaderc::Compiler& compiler,
-                        const std::string& shader, shaderc_shader_kind kind,
-                        const shaderc::CompileOptions& options) {
-  shaderc::SpvModule result = compiler.CompileGlslToSpv(shader, kind, options);
-  if (!result.GetSuccess()) return false;
-  size_t length = result.GetLength();
-  if (length < 20) return false;
-  const uint32_t* bytes =
-      static_cast<const uint32_t*>(static_cast<const void*>(result.GetData()));
-  return bytes[0] == spv::MagicNumber;
-}
-
-TEST(CppInterface, CompilerMoves) {
-  shaderc::Compiler compiler;
-  ASSERT_TRUE(compiler.IsValid());
-  shaderc::Compiler compiler2(std::move(compiler));
-  ASSERT_FALSE(compiler.IsValid());
+TEST_F(CppInterface, CompilerMoves) {
+  ASSERT_TRUE(compiler_.IsValid());
+  shaderc::Compiler compiler2(std::move(compiler_));
+  ASSERT_FALSE(compiler_.IsValid());
   ASSERT_TRUE(compiler2.IsValid());
 }
 
-TEST(CppInterface, EmptyString) {
-  shaderc::Compiler compiler;
-  ASSERT_TRUE(compiler.IsValid());
-  EXPECT_FALSE(CompilationSuccess(compiler, "", shaderc_glsl_vertex_shader));
-  EXPECT_FALSE(CompilationSuccess(compiler, "", shaderc_glsl_fragment_shader));
+TEST_F(CppInterface, EmptyString) {
```

Comment:
It's something I noticed while reviewing this PR.

Agreed, it's primarly a DRYness change.